### PR TITLE
Update libraries

### PR DESCRIPTION
**Pull request by saucecontrol**
Yet another update 😅 

First commit
1) updates libraries to current versions, *except*
2) downgrades System.Drawing.Common, which is marked as Windows-only in 6.0+ (with no functional changes) leading to tons of unnecessary warnings
3) adds a check to the remaining Windows-only calls to satisfy the platform compat analyzer
3) updates MagicScaler benchmark for API changes in the latest version
4) adds a missing `using` to the FreeImage benchmark
5) removes AssemblyInfo.cs and the .csproj properties that were included to suppress errors for duplicate atttributes

Second commit removes the SkiaSharp Canvas benchmarks.  Skia has had only one way to resize [since v1.68](https://github.com/mono/SkiaSharp/issues/520#issuecomment-442923077), so these were just burning extra time and always giving the same results.  The canvas samples were convoluted and had greater managed allocations, which is why I chose them to remove.

Results of the latest run on my Tiger Lake laptop (before the Skia benchmark consolidation)

``` ini
BenchmarkDotNet=v0.13.1.1695-nightly, OS=Windows 10 (10.0.19043.1586/21H1/May2021Update)
Intel Xeon W-11855M CPU 3.20GHz, 1 CPU, 12 logical and 6 physical cores
.NET SDK=6.0.300-preview.22204.3
  [Host]   : .NET 6.0.3 (6.0.322.12309), X64 RyuJIT
  ShortRun : .NET 6.0.3 (6.0.322.12309), X64 RyuJIT
```

```
|                              Method |      Mean |     Error |   StdDev | Ratio |     Gen 0 |     Gen 1 |     Gen 2 |  Allocated | Alloc Ratio |
|------------------------------------ |----------:|----------:|---------:|------:|----------:|----------:|----------:|-----------:|------------:|
|   System.Drawing Load, Resize, Save | 349.95 ms |  2.492 ms | 0.647 ms |  1.00 |         - |         - |         - |   10.23 KB |        1.00 |
|       ImageSharp Load, Resize, Save | 106.31 ms | 10.990 ms | 2.854 ms |  0.30 |  156.2500 |   62.5000 |         - | 1492.71 KB |      145.98 |
|      ImageMagick Load, Resize, Save | 340.18 ms | 18.236 ms | 2.822 ms |  0.97 |         - |         - |         - |   51.75 KB |        5.06 |
|        ImageFree Load, Resize, Save | 210.36 ms |  2.737 ms | 0.711 ms |  0.60 | 6000.0000 | 6000.0000 | 6000.0000 |   92.69 KB |        9.06 |
|      MagicScaler Load, Resize, Save |  55.45 ms |  2.228 ms | 0.579 ms |  0.16 |         - |         - |         - |   45.51 KB |        4.45 |
| SkiaSharp Canvas Load, Resize, Save | 168.27 ms | 12.236 ms | 3.178 ms |  0.48 |         - |         - |         - |  101.44 KB |        9.92 |
| SkiaSharp Bitmap Load, Resize, Save | 164.13 ms | 10.054 ms | 2.611 ms |  0.47 |         - |         - |         - |    85.5 KB |        8.36 |
|          NetVips Load, Resize, Save | 101.94 ms |  5.045 ms | 0.781 ms |  0.29 |         - |         - |         - |   46.04 KB |        4.50 |
```

```
|                                         Method |      Mean |    Error |   StdDev | Ratio |     Gen 0 |     Gen 1 |     Gen 2 |  Allocated | Alloc Ratio |
|----------------------------------------------- |----------:|---------:|---------:|------:|----------:|----------:|----------:|-----------:|------------:|
|   System.Drawing Load, Resize, Save - Parallel | 133.11 ms | 0.709 ms | 0.184 ms |  1.00 |         - |         - |         - |   39.41 KB |        1.00 |
|       ImageSharp Load, Resize, Save - Parallel |  22.91 ms | 0.281 ms | 0.073 ms |  0.17 |  156.2500 |   78.1250 |         - | 1524.18 KB |       38.68 |
|      ImageMagick Load, Resize, Save - Parallel | 100.92 ms | 5.891 ms | 1.530 ms |  0.76 |         - |         - |         - |   85.45 KB |        2.17 |
|        ImageFree Load, Resize, Save - Parallel |  46.08 ms | 2.399 ms | 0.371 ms |  0.35 | 3062.5000 | 3062.5000 | 3062.5000 |  118.96 KB |        3.02 |
|      MagicScaler Load, Resize, Save - Parallel |  11.49 ms | 0.643 ms | 0.100 ms |  0.09 |         - |         - |         - |   73.98 KB |        1.88 |
| SkiaSharp Canvas Load, Resize, Save - Parallel |  34.09 ms | 0.527 ms | 0.082 ms |  0.26 |   15.6250 |         - |         - |  134.99 KB |        3.43 |
| SkiaSharp Bitmap Load, Resize, Save - Parallel |  34.36 ms | 1.025 ms | 0.159 ms |  0.26 |   15.6250 |         - |         - |   116.9 KB |        2.97 |
|          NetVips Load, Resize, Save - Parallel |  38.69 ms | 0.252 ms | 0.065 ms |  0.29 |         - |         - |         - |   77.99 KB |        1.98 |
```

```
|                  Method |        Mean |     Error |    StdDev | Ratio | RatioSD |    Gen 0 |    Gen 1 |    Gen 2 | Allocated | Alloc Ratio |
|------------------------ |------------:|----------:|----------:|------:|--------:|---------:|---------:|---------:|----------:|------------:|
|   System.Drawing Resize |  9,093.6 μs |  84.16 μs |  74.61 μs |  1.00 |    0.00 |        - |        - |        - |     138 B |        1.00 |
|       ImageSharp Resize |  2,247.4 μs |  94.39 μs |  88.29 μs |  0.25 |    0.01 |        - |        - |        - |   10364 B |       75.10 |
|      ImageMagick Resize | 37,032.8 μs | 906.42 μs | 847.87 μs |  4.08 |    0.11 |        - |        - |        - |    5338 B |       38.68 |
|        FreeImage Resize |  5,940.9 μs | 119.98 μs | 112.23 μs |  0.65 |    0.01 | 500.0000 | 500.0000 | 500.0000 |     306 B |        2.22 |
|      MagicScaler Resize |    577.9 μs |   1.01 μs |   0.94 μs |  0.06 |    0.00 |        - |        - |        - |    1425 B |       10.33 |
| SkiaSharp Canvas Resize |  1,577.6 μs |  19.76 μs |  18.49 μs |  0.17 |    0.00 |        - |        - |        - |    1745 B |       12.64 |
| SkiaSharp Bitmap Resize |  1,558.8 μs |  10.30 μs |   9.13 μs |  0.17 |    0.00 |        - |        - |        - |     489 B |        3.54 |
|          NetVips Resize |  3,540.6 μs |  23.28 μs |  21.78 μs |  0.39 |    0.00 |        - |        - |        - |    3859 B |       27.96 |
```

### CODE REVIEW

**Comment by saucecontrol** [comment]
@JimBobSquarePants it looks like you've added XMP and IPTC metadata support in ImageSharp 2.x, and that's making the thumbnail files about 3x larger than they used to be.  Do you have a simple way to preserve ICC profile but discard the remaining metadata?

Same question for @kleisauke.  I think I asked before if NetVips can preserve color profile when the remaining metadata is discarded, but I don't remember seeing an answer.  Is that possible?

File: NetCore/LoadResizeSave.cs [hunk 1]
```diff
@@ -244,51 +246,13 @@ public void MagicScalerResize(string input)
...
 
-            using (var output = new FileStream(OutputPath(input, MagicScaler), FileMode.Create))
-            {
-                MagicImageProcessor.ProcessImage(input, output, settings);
-            }
-        }
-
-        [Benchmark(Description = "SkiaSharp Canvas Load, Resize, Save"/*,
-            OperationsPerInvoke = ImagesCount*/)]
-        public void SkiaCanvasBenchmark()
```
Comment (kleisauke) [inline NetCore/LoadResizeSave.cs:260]: This deleted function is still referenced in `Program.cs`.
https://github.com/bleroy/core-imaging-playground/blob/5bbfba070a161a3f5c4b77799e6f4a8fce3b6d48/NetCore/Program.cs#L122

File: NetCore/LoadResizeSave.cs [hunk 2]
```diff
@@ -244,51 +246,13 @@ public void MagicScalerResize(string input)
...
-        {
-            foreach (string image in Images)
-            {
-                SkiaCanvasResize(image);
-            }
-        }
-
-        public void SkiaCanvasResize(string input)
```
Comment (kleisauke) [inline NetCore/LoadResizeSave.cs:268]: This deleted function is still referenced in `LoadResizeSaveParallel.cs`.
https://github.com/bleroy/core-imaging-playground/blob/5bbfba070a161a3f5c4b77799e6f4a8fce3b6d48/NetCore/LoadResizeSaveParallel.cs#L47

**Comment by JimBobSquarePants** [comment]
@saucecontrol yeah, you can set each profile to `null`.

Beginning to regret XMP support, that one’s caught out a few people.

**Comment by kleisauke** [comment]
For NetVips, to preserve just the ICC profile, you can do:

```diff
--- a/NetCore/LoadResizeSave.cs
+++ b/NetCore/LoadResizeSave.cs
@@ -299,8 +299,17 @@ namespace ImageProcessing
             // Thumbnail to fit a 150x150 square
             using (var thumb = NetVipsImage.Thumbnail(input, ThumbnailSize, ThumbnailSize))
             {
+                using var mutated = thumb.Mutate(mutable =>
+                {
+                    foreach (var field in mutable.GetFields())
+                    {
+                        if (field == "icc-profile-data") continue;
+                        mutable.Remove(field);
+                    }
+                });
+
                 // Save the results
-                thumb.Jpegsave(OutputPath(input, NetVips), q: Quality, strip: true);
+                mutated.Jpegsave(OutputPath(input, NetVips), q: Quality);
             }
         }
     }
```

Or, for transforming to sRGB (which might be more appropriate), you could do:

```diff
--- a/NetCore/LoadResizeSave.cs
+++ b/NetCore/LoadResizeSave.cs
@@ -297,7 +297,13 @@ namespace ImageProcessing
         public void NetVipsResize(string input)
         {
             // Thumbnail to fit a 150x150 square
-            using (var thumb = NetVipsImage.Thumbnail(input, ThumbnailSize, ThumbnailSize))
+            using (var thumb = NetVipsImage.Thumbnail(input, ThumbnailSize, ThumbnailSize,
+                       // Fallback to sRGB if the image has no embedded ICC profile
+                       importProfile: "srgb",
+                       // Transform to sRGB
+                       exportProfile: "srgb",
+                       // Use "perceptual" intent to better match IM
+                       intent: Enums.Intent.Perceptual))
             {
                 // Save the results
                 thumb.Jpegsave(OutputPath(input, NetVips), q: Quality, strip: true);
```

@jcupitt I've seen this question several times now. Perhaps we could change the strip option to support flags (so the user has more control over this)? Just like [`VipsForeignPngFilter`](https://github.com/libvips/libvips/blob/07edfe37b18c05a24b235331939dfce8b865bca4/libvips/include/vips/foreign.h#L669-L688), for example in C#:
```csharp
[Flags]
public enum ForeignStrip
{
    /// <summary>No stripping.</summary>
    None = 0, // "none"

    /// <summary>Strip EXIF metadata.</summary>
    Exif = 1 << 0, // "exif"

    /// <summary>Strip XMP metadata.</summary>
    Xmp = 1 << 1, // "xmp"

    /// <summary>Strip ICC metadata.</summary>
    Icc = 1 << 2, // "icc"

    /// <summary>Strip IPTC metadata.</summary>
    Iptc = 1 << 3, // "iptc"

    /// <summary>Strip all metadata.</summary>
    All = Exif | Xmp | Icc | Iptc // "all"
}
```
```csharp
// Save as JPEG while only preserving the ICC profile
thumb.Jpegsave(OutputPath(input, NetVips), q: Quality, strip: Enums.ForeignStrip.All & ~Enums.ForeignStrip.Icc);
```

**Comment by jcupitt** [comment]
Good idea, though I guess we couldn't call it `strip`. Maybe flip the sense and call it `preserve`, then deprecate `strip`?

```
vips copy x.jpg y.tif[preserve=icc|exif]
```

File: NetCore/LoadResizeSave.cs (hunk 1 above)
Comment (saucecontrol) [reply NetCore/LoadResizeSave.cs:260]: D'oh, that's what I get for submitting PRs late at night.

Fixed.

**Comment by saucecontrol** [comment]
> @saucecontrol yeah, you can set each profile to `null`.

Ha, I didn't notice that was already being done for Exif.  I've updated to set XMP and IPTC to null as well, and sizes are back to normal.  Also noticed ImageSharp was saving with 4:4:4 subsampling (did the default change at some point?) and set that explicitly to 4:2:0

> For NetVips, to preserve just the ICC profile, you can do:

I went with the explicit stripping since color conversion would be a perf hit.  Skia switched at some point to doing sRGB conversion by default, and their numbers nearly doubled.

**Comment by bleroy** [comment]
Thanks for the contribution!

**Comment by JimBobSquarePants** [comment]
>Also noticed ImageSharp was saving with 4:4:4 subsampling (did the default change at some point?) and set that explicitly to 4:2:0

@saucecontrol Yeah, we now set subsampling based on quality only when the input image does not contain subsampling metadata in order to preserve as much of the original visual data as possible.

**Pull request merged**